Flow is not compatible with JDK 21 early access

### Description of the bug

building flow with JDK 21 early access is not successful

```
[ERROR] Failed to execute goal org.jboss.jandex:jandex-maven-plugin:1.1.1:jandex (make-index) on project flow-jandex: Execution make-index of goal org.jboss.jandex:jandex-maven-plugin:1.1.1:jandex failed: Cannot read the array length because "bytes" is null -> [Help 2]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.jboss.jandex:jandex-maven-plugin:1.1.1:jandex (make-index) on project flow-jandex: Execution make-index of goal org.jboss.jandex:jandex-maven-plugin:1.1.1:jandex failed: Cannot read the array length because "bytes" is null
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:196)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:186)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)

```

### Expected behavior

build is successful with JDK 21 (official release on 19th Sept 2023)

### Minimal reproducible example

- take the current flow main branch
- set java_home to the JDK 21 (Java version: 21-ea, vendor: Oracle Corporation)
- run `mvn -e -B -V -T 6 -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn install -DskipTests `

### Versions

- Java version: openjdk-21-ea+31

--- a/flow-jandex/pom.xml
+++ b/flow-jandex/pom.xml
@@ -102,9 +102,9 @@
                 </executions>
             </plugin>
             <plugin>
-                <groupId>org.jboss.jandex</groupId>
+                <groupId>io.smallrye</groupId>
                 <artifactId>jandex-maven-plugin</artifactId>
-                <version>1.1.1</version>
+                <version>3.1.2</version>
                 <executions>
                     <execution>
                         <id>make-index</id>

